hbase-hadoop2-compat/src/main/java/org/apache/hadoop/hbase/replication/regionserver/MetricsReplicationSinkSourceImpl.java

```diff
@@ -27,12 +27,14 @@ public class MetricsReplicationSinkSourceImpl implements MetricsReplicationSinkS
 
   private final MutableHistogram ageHist;
   private final MutableFastCounter batchesCounter;
+  private final MutableFastCounter failedBatchesCounter; // TODO(baugenreich) mutable gauge long instead??
```

Comment:
I think counter makes sense -- gauges make sense for a point in time, but if you want to check things over time counter is better. This way we can look at delta, rates, and totals. One guage-like metric I could imagine here would be "number of batches currently retrying" or something like that, since it's measuring somethign that's actively happening (retry backoffs maybe).